packages/ngtools/webpack/src/ivy/plugin.ts

```diff
@@ -0,0 +1,488 @@
+      compilerOptions.outDir = undefined;
+      compilerOptions.inlineSources = compilerOptions.sourceMap;
+      compilerOptions.inlineSourceMap = false;
+      compilerOptions.mapRoot = undefined;
+      compilerOptions.sourceRoot = undefined;
+      compilerOptions.annotationsAs = 'decorators';
+
+      // Create diagnostics reporter and report configuration file errors
+      const diagnosticsReporter = createDiagnosticsReporter(compilation);
+      diagnosticsReporter(errors);
+
+      // Update TypeScript path mapping plugin with new configuration
+      pathsPlugin.update(compilerOptions);
+
+      // Create a Webpack-based TypeScript compiler host
+      const system = createWebpackSystem(
+        compiler.inputFileSystem,
+        forwardSlashPath(compiler.context),
+      );
+      const host = ts.createIncrementalCompilerHost(compilerOptions, system);
+
+      // Setup source file caching and reuse cache from previous compilation if present
+      let cache = this.sourceFileCache;
+      if (cache) {
+        // Invalidate existing cache based on compilation file timestamps
+        for (const [file, time] of compilation.fileTimestamps) {
+          if (this.buildTimestamp < time) {
+            cache.delete(forwardSlashPath(file));
+          }
+        }
+      } else {
+        // Initialize a new cache
+        cache = new Map();
+        // Only store cache if in watch mode
+        if (this.watchMode) {
+          this.sourceFileCache = cache;
+        }
+      }
+      this.buildTimestamp = Date.now();
+      augmentHostWithCaching(host, cache);
+
+      const moduleResolutionCache = ts.createModuleResolutionCache(
+        host.getCurrentDirectory(),
+        host.getCanonicalFileName.bind(host),
+        compilerOptions,
+      );
+
+      // Setup on demand ngcc
+      augmentHostWithNgcc(host, ngccProcessor, moduleResolutionCache);
+
+      // Setup resource loading
+      resourceLoader.update(compilation);
+      augmentHostWithResources(host, resourceLoader, {
+        directTemplateLoading: this.pluginOptions.directTemplateLoading,
+      });
+
+      // Setup source file adjustment options
+      augmentHostWithReplacements(host, this.pluginOptions.fileReplacements, moduleResolutionCache);
+      augmentHostWithSubstitutions(host, this.pluginOptions.substitutions);
+
+      // Create the file emitter used by the webpack loader
+      const { fileEmitter, builder } = compilerOptions.skipTemplateCodegen
+        ? this.updateJitProgram(compilerOptions, rootNames, host, diagnosticsReporter)
+        : this.updateAotProgram(
+            compilerOptions,
+            rootNames,
+            host,
+            diagnosticsReporter,
+            resourceLoader,
+          );
+
+      const allProgramFiles = builder
+        .getSourceFiles()
+        .map((sourceFile) => sourceFile.fileName)
+        .filter(
+          (sourceFile) =>
+            !/(\.(ngfactory|ngstyle|ngsummary|ngtypecheck)\.ts)$/.test(sourceFile) &&
```

Comment:
You seem to have forgotten to remove this line
```suggestion
            !/(\.(ngfactory|ngstyle|ngsummary|ngtypecheck)\.ts)$/.test(sourceFile) &&
```


ngtypecheck -> was used before
ngfactory|ngstyle|ngsummary -> are for VE